### ISSUE TITLE
Add GBSA support for GAFF residue template genertors

### ISSUE THREAD

**Issue by jchodera** [openmm/openmm-forcefields#84]
We should add GBSA support for GAFF residue template generators.

**Comment by chuantian** [comment]
I'm wondering if people plan to include GB models (xml file) such as OBC1/2, neck, neck2 etc into openmmforcefields directory.

**Comment by jchodera** [comment]
Unlike the `openforcefield` approach, OpenMM's `ForceField` ties GB parameters to individual atom types or classes, which creates a huge headache: It essentially requires that every atom of every residue or small molecule have a unique atom type since the GB atom types and force field atom types do not necessarily correspond.

Recently, @swails added code to [assign parameters for various GBSA models to OpenMM `Topology` objects](https://github.com/openmm/openmm/blob/master/wrappers/python/simtk/openmm/app/internal/customgbforces.py)---see, for example [`simtk.openmm.app.internal.GBSAGBnForce.getStandardParameters(topology)`](https://github.com/openmm/openmm/blob/master/wrappers/python/simtk/openmm/app/internal/customgbforces.py#L752)---but this code is currently only used to _check_ if recommended parameters have been loaded.

We could repurpose this code within `openmmforcefields` to provide a way to assign GB parameters after a vacuum system has been constructed, but we would need to add some API for assigning GB parameters after the fact. How would we want that to look, and what would the use cases be?

**Comment by jchodera** [comment]
@peastman : To support GBSA parameters for new-style force field conversions, would it work for me to simply create GBSA XML files (e.g.`obc2_gbsa.xml`) containing a `<CustomGBGenerator>` and a `<Script>` that uses [`simtk.openmm.app.internal.customgbforces.GBSAOBC2Force.getStandardParameters(topology)`](https://github.com/openmm/openmm/blob/master/wrappers/python/simtk/openmm/app/internal/customgbforces.py#L622) to retrieve and assign GBSA parameters for the whole system? Could this work in general, for both converted biopolymers and small molecules?

**Comment by peastman** [comment]
Yes, that should work.  You could even omit the `<CustomGBForce>` tag and just have the script create the force directly.

**Comment by jchodera** [comment]
cc https://github.com/openmm/openmm/issues/2675 since this is better handled within OpenMM itself!
